server.go

```diff
@@ -296,19 +305,53 @@ type Server struct {
 	DecorateReader DecorateReader
 	// DecorateWriter is optional, allows customization of the process that writes raw DNS messages.
 	DecorateWriter DecorateWriter
+	// Graceful shutdown handling
+	running int32
+	// Messages queue
+	queue chan *message
+	// Number of workers, if set to zero - use spawn goroutines instead
+	Workers int
+	// Size of DNS requests queue
+	QueueSize int
```

Comment:
the #workers is one thing, but the QueueSize is quite another, if this doesn't perform better with QueueSize == 0 it's not worth adding. A buffered channel leads to weird "sometimes it is slow - or blocking" errors in prod when the thing finally fills up.